### ISSUE TITLE
Invalid Signature

### ISSUE THREAD

**Issue by digitalrbnz** [jonasroussel/dart_jsonwebtoken#58]
I am creating a client assertion signed JWT using RSA and  PS512. The token seems to be fine if decoded and all the signatures verify in JWT IO and state the correct algorithm. 

However when I attempt to use this as part of a client credential auth, I get an invalid signature error. 

If I use exactly the same key kid, claims etc using KJUR library in js or postman, it works fine.

One thing I noticed is that the token size using KJUR us much smaller than using this library, even though the header, payment and and signing are all the same.

Are there any known issues here or something obvious I might be missing?

**Comment by user163** [comment]
The problem is that the library for PSS uses a fixed salt length of 32 bytes (s. [here](https://github.com/jonasroussel/dart_jsonwebtoken/blob/v2.14.1/lib/src/algorithms.dart#L213)), while [RFC 7518, sec. 3.5](https://datatracker.ietf.org/doc/html/rfc7518#section-3.5) defines that the salt length must correspond to the digest output size, i.e. 32 bytes for PS256, 48 bytes for PS384 and 64 bytes for PS512. Therefore, the current implementation only generates RFC7518 compliant signatures for PS256, but neither for PS384 nor for PS512.

**Comment by digitalrbnz** [comment]
> The problem is that the dart library for PSS uses a fixed salt length of 32 bytes (s. [here](https://github.com/jonasroussel/dart_jsonwebtoken/blob/v2.14.1/lib/src/algorithms.dart#L213)), while [RFC 7518, sec. 3.5](https://datatracker.ietf.org/doc/html/rfc7518#section-3.5) defines that the salt length must correspond to the digest output size, i.e. 32 bytes for PS256, 48 bytes for PS384 and 64 bytes for PS512. Therefore, the current implementation only generates RFC7518 compliant signatures for PS256, but neither for PS384 nor for PS512.

Thanks for the response.  Surely this makes the library usable for anything other than PS256? Is there a fix for this at all?

**Comment by digitalrbnz** [comment]
@user163 Agreed. That was just for testing purposes. I have added a switch statement to the sign and verify functions and preliminary testing checks out. PR has been raised -> https://github.com/jonasroussel/dart_jsonwebtoken/pull/59

Thanks again for identifying the issue. You have been credited in the PR.

**Comment by digitalrbnz** [comment]
> [@digitalrbnz](https://github.com/digitalrbnz) Problem fixed! Your changes were correct, the only problem came from the secure random that was seeded before passing it to the PSS parameters. It was an error because PointyCastle already does it correctly with the salt information contained inside the signature.
> 
> PR merged and released under this version : https://pub.dev/packages/dart_jsonwebtoken/versions/2.15.0

Thanks for that and thanks for providing a great library.
